### PR TITLE
Gtask command queue

### PR DESCRIPTION
**Pull request by keis**
First part of a bigger effort to allow commands to be implemented using GTask which should let us implement #153 without fighting the new webkit api too much. It has proven much easier to port this from the outer layer and moving in rather than starting with the actual command implementation. However I have a pending patch that I think will mostly work if put on top of this.

Dropping the custom source in favour of a task with a blocking pop call in a thread. This should still be reasonably fast because gio uses a shared thread pool.

### CODE REVIEW

File: src/io.c [hunk 1]
```diff
@@ -59,9 +58,13 @@ flush_event_buffer (gpointer data);
 static void
 free_cmd_req (gpointer data);
```
Comment (mathstuf) [inline src/io.c]: Can this be removed now?

**Comment by mathstuf** [comment]
I like :) . Travis seems to have had an intermittent failure. Not sure how to kick it to retry just that build though.

File: src/io.c (hunk 1 above)
Comment (keis) [inline src/io.c]: The struct is still around just a bit smaller. It's used as the task data (g_task_set_task_data)

**Comment by keis** [comment]
Rerunning that job now. There is this small circle arrow button in the summary block if you open the specific job

**Comment by mathstuf** [comment]
Ah, that probably requires logging in.

**Comment by mathstuf** [comment]
LGTM.

**Comment by keis** [comment]
Ok, cool. I'll go ahead and merge

**Pull request merged**